sisl/grid.py

```diff
@@ -180,6 +180,68 @@ def interp(self, shape, order=1, mode="wrap", **kwargs):
         # Apply the scipy.ndimage.zoom function and return a new grid
         return self.apply(ndimage_zoom, zoom_factors, mode=mode, order=order, **kwargs)
 
+    def isosurface(self, level=None, step_size=1, **kwargs):
+        """Calculates the isosurface for a given value.
+        
+        It uses skimage.measure.marching_cubes, so you need to have scikit-image installed.
+
+        Parameters
+        ----------
+        level: float
+            contour value to search for isosurfaces in the grid. 
+            If not given or None, the average of the min and max of the grid is used.
+        step_size: int
```

Comment:
I think this is not needed, it will already be handled by `kwargs` and I guess (?) that users will tend to want the same resolution?

Reply:
I was not sure about this, because I think the typical resolution of a grid in SIESTA results in a very big amount of vertices and faces, which can be hard for the rendering engines to display (depending on the computer of course). Sometimes they don't add any visual information versus a surface that is defined with less resolution. Of course the default needs to be 1, but by putting it here I wanted to emphasize that you can tune the parameter, so that no one gets stuck with a too large number of vertices.

If you think this is not important or that it is enough to point to the docs of `skimage.measure.marching_cubes` I will remove it.

Reply:
Ok, agreed, keep it.